src/bson/helpers/builder.cr

```diff
@@ -121,11 +121,41 @@ struct BSON
       @io.write_bytes 0x00_u8, IO::ByteFormat::LittleEndian
     end
 
+    def []=(key : String, value : UInt8)
+      field(:int32, key)
+      @io.write_bytes value.to_i32, IO::ByteFormat::LittleEndian
+    end
+
+    def []=(key : String, value : Int8)
+      field(:int32, key)
+      @io.write_bytes value.to_i32, IO::ByteFormat::LittleEndian
+    end
+
+    def []=(key : String, value : UInt16)
+      field(:int32, key)
+      @io.write_bytes value.to_i32, IO::ByteFormat::LittleEndian
+    end
+
+    def []=(key : String, value : Int16)
+      field(:int32, key)
+      @io.write_bytes value.to_i32, IO::ByteFormat::LittleEndian
+    end
+
+    def []=(key : String, value : UInt32)
+      field(:int32, key)
+      @io.write_bytes value.to_i32, IO::ByteFormat::LittleEndian
```

Comment:
`UInt32` should be cast to `Int64` instead of `Int32` (max UInt32 value > max Int32 value)

Reply:
Fixed in following commit : 26d497b85f33577faeea86a7c2519ae42afa8869